project_scrum/models/project.py

```diff
@@ -33,10 +33,14 @@ class ProjectSprint(models.Model):
     is_current_sprint = fields.Boolean('Is Current Sprint')
     is_previous_sprint = fields.Boolean('Is Previous Sprint')
 
-    @api.one
     def _task_count(self):
-        tasks = self.env['project.task'].search([('sprint_id', '=', self.id)])
-        self.task_count = len(tasks)
+        tasks = self.env['project.task'].read_group(
+            [('sprint_id', 'in', self.ids)],
+            ['sprint_id'],
+            ['sprint_id'])
+        for r in self:
+            task_group = [group for group in tasks if group['sprint_id'][0] == r.id]
```

Comment:
We're already grouping by `sprint_id`, shouldn't that guarantee that we only see each `sprint_id` only once? This looks like worst case we'll loop through all the groups as many times as we have groups to go through. 

Seems like it would be more efficient to extract the data into a lookup dictionary first:
```python
sprint_counts = {group['sprint_id'][0]: group['sprint_id_count'] for group in tasks}
for sprint in self:
    sprint.task_count = sprint_counts.get(sprint.id, 0)
```

Reply:
I made change you suggested.